x-pack/plugins/security/public/management/role_mappings/edit_role_mapping/rule_editor_panel/json_rule_editor.tsx

```diff
@@ -76,25 +77,26 @@ export const JSONRuleEditor = (props: Props) => {
       data-test-subj="roleMappingsJSONEditor"
     >
       <Fragment>
-        <EuiCodeEditor
+        <CodeEditor
           aria-label={''}
-          mode={'json'}
-          theme="github"
+          languageId={XJsonLang.ID}
           value={rawRules}
           onChange={onRulesChange}
-          width="100%"
-          height="auto"
-          minLines={6}
-          maxLines={30}
-          isReadOnly={false}
-          setOptions={{
-            showLineNumbers: true,
+          fullWidth={true}
+          height="300px"
```

Comment:
This component doesn't support `maxLines`/`minLines` but it needs a height, so I just set it to 300px which seems reasonable (which equals around 17 lines), right in the middle).